path_login.go

```diff
@@ -401,7 +407,33 @@ func (b *azureAuthBackend) verifyResource(ctx context.Context, subscriptionID, r
 	// Ensure the token OID is the principal id of the system-assigned identity
 	// or one of the user-assigned identities
 	if _, ok := principalIDs[claims.ObjectID]; !ok {
-		return errors.New("token object id does not match expected identities")
+		// if it isn't, check the appID and see if _that_ exists.
```

Comment:
Suggest to expand on this comment to explain any non-obvious details that might be easy to forget in the future. For example, it might be worth mentioning that this fallback validation is in place to support WIF and touch on what's being validated.

Reply:
added some words!